server/leafnode.go

```diff
@@ -1292,10 +1285,49 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 	key := string(sub.sid)
 	osub := c.subs[key]
 	updateGWs := false
+	acc := c.acc
 	if osub == nil {
+		subj := string(sub.subject)
+		accUnlock := false
+		// Check if we have a loop.
+		if len(subj) >= len(leafNodeLoopDetectionSubjectPrefixOld) {
+			subStripped := subj
+			if subStripped[0] == '$' {
+				subStripped = subStripped[1:]
+			}
+
```

Comment:
Just "code convention": we tend not to add blank lines that often. Nowadays, with IDEs and code coloring, the code and comments can be easily distinguished.